File: snippets/csharp/VS_Snippets_CLR/conceptual.basicio.textfiles/cs/source6.cs
// <snippet6>
using System;
using System.IO;
using System.Windows;

namespace WpfApp2
{
    /// <summary>
    /// Interaction logic for MainWindow.xaml
    /// </summary>
    public partial class MainWindow : Window
    {
        public MainWindow()
        {
            InitializeComponent();
        }
        private async void ReadFileButton_Click(object sender, RoutedEventArgs e)
        {
            try
            {
                using (StreamReader sr = new StreamReader("TestFile.txt"))
                {
                    string line = await sr.ReadToEndAsync();
                    Console.WriteLine(line);
                }
            }
            catch (FileNotFoundException ex)
            {
                Console.WriteLine(ex.Message);
            }
        }
    }
}
// </snippet6>

File: snippets/visualbasic/VS_Snippets_CLR/conceptual.basicio.textfiles/vb/source6.vb
' <snippet6>
Imports System.IO
Imports System.Windows

Class MainWindow
    Private Async Sub ReadFileButton_Click(sender As Object, e As RoutedEventArgs)
        Try
            Using sr As StreamReader = New StreamReader("TestFile.txt")
                Dim line = Await sr.ReadToEndAsync()
                Console.WriteLine(line)
            End Using
        Catch ex As FileNotFoundException
            Console.WriteLine(ex.Message)
        End Try
    End Sub
End Class
' </snippet6>